Use github docker container

tiziodcaio opened the pull request:
Cleaner setup

tiziodcaio committed:
Use github docker container

Cleaner setup

plfiorini reviewed (changes requested):
Why not just:

```
runs-on: ubuntu-latest
```

The build script should run on future Ubuntu versions as well.

tiziodcaio committed:
Update checkout actions

tiziodcaio committed:
Fix workflow

Using the 22.10 system might be used only with the previous system
Warning! Using ubuntu-latest will downgrade to 20.04

plfiorini commented:
Maybe the "Dependencies" step needs sudo in every line, can you see if that works?

tiziodcaio committed:
Update build.yml

plfiorini committed:
sudo

aleixpol commented:
I'm not sure it's a good idea to be using an older version of the toolchain. I am not sure what this change brings right now.
